### ISSUE TITLE
Add use of proxy to startup messages

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#681]
**Reported by Ageless on 8 Jul 38577827 21:20 UTC**
At this moment any client that starts up shows about everything that the client can do and has set as options and settings. I'm requesting that a notification that a proxy is in use is added to this. 

So that it shows something like this:[BOINC client version x.x.x for platform[[br]([br]]
Starting)]
log flags: task, file_xfer, sched_ops[libcurl/7.18.0 OpenSSL/0.9.8e zlib/1.2.3[[br]([br]]
Libraries:)]
Data directory: C:\Program Files\BOINC[processor detection[[br]([br]]
Processor:)]
Processor features: fpu tsc pae nx sse sse2 mmx[Operating system[[br]([br]]
OS:)]
Memory: x.00 GB physical, x.xx GB virtual[xx GB total, xx GB free[[br]([br]]
Disk:)]
Local time is UTC +x hours[coprocessors[[br]([br]]
No)]
'''Proxy used: None/HTTP/Socks'''[[br]]

Where it shows None when none is in use and either HTTP or Socks depending on which one is in use. It helps us when diagnosing people's problems and they post their messages.

**Comment by romw** [comment]
**Commented by Nicolas on 1 Mar 38578847 20:53 UTC**
+1

**Comment by romw** [comment]
**Commented by Pepo on 25 Nov 38580362 23:33 UTC**
Replying to [Ageless](ticket:704):
> '''Proxy used: None/HTTP/Socks'''[[br]]
> 
The notification might then also appear each time the proxy setting is changed on the fly. (I'm aware that the message will possibly disappear among other messages.)

**Comment by romw** [comment]
**Commented by davea on 10 Feb 38581547 03:06 UTC**
(In [15716]) - client: show proxy info in messages at startup and when changed
    fixes #704
- web: fix the am_set_info.php RPC
    (most args are optional, not mandatory)

**Comment by romw** [comment]
**Commented by Pepo on 1 May 38581608 05:46 UTC**
Replying to [davea](comment:3):
> (In [- client: show proxy info in messages at startup and when changed
>   [[br](15716]))]  fixes #704 
Switching off the proxy (or not using it at the beginning) is not reported.

Why not report all states, something like
```C
void CLIENT_STATE::show_proxy_info() {
    if (proxy_info.use_http_proxy) {
        msg_printf(NULL, MSG_INFO, "Using HTTP proxy %s:%d",
            proxy_info.http_server_name, proxy_info.http_server_port
        );
    }
    else if (proxy_info.use_socks_proxy) {
        msg_printf(NULL, MSG_INFO, "Using SOCKS proxy %s:%d",
             proxy_info.socks_server_name, proxy_info.socks_server_port
        );
    }
    else {
        msg_printf(NULL, MSG_INFO, "Not using a proxy");
    }
}
```

**Comment by romw** [comment]
**Commented by davea on 23 Nov 38581770 05:46 UTC**
(In [15717]) - client: show message if no proxy
    fixes #704
